platform/features/table/src/controllers/HistoricalTableController.js

```diff
@@ -35,32 +36,79 @@ define(
          * @param telemetryFormatter
          * @constructor
          */
-        function HistoricalTableController($scope, telemetryHandler, telemetryFormatter) {
+        function HistoricalTableController($scope, telemetryHandler, telemetryFormatter, $timeout) {
+            var self = this;
+
+            this.$timeout = $timeout;
+            this.timeoutHandle = undefined;
+            this.batchSize = BATCH_SIZE;
+
+            $scope.$on("$destroy", function () {
+                clearTimeout(self.timeoutHandle);
+            });
+
             TableController.call(this, $scope, telemetryHandler, telemetryFormatter);
         }
 
         HistoricalTableController.prototype = Object.create(TableController.prototype);
 
         /**
-         * Populates historical data on scope when it becomes available from
-         * the telemetry API
-         */
+        * Populates historical data on scope when it becomes available from
+        * the telemetry API
+        */
         HistoricalTableController.prototype.addHistoricalData = function () {
             var rowData = [],
-                self = this;
+                self = this,
+                telemetryObjects = this.handle.getTelemetryObjects();
 
-            this.handle.getTelemetryObjects().forEach(function (telemetryObject) {
-                var series = self.handle.getSeries(telemetryObject) || {},
-                    pointCount = series.getPointCount ? series.getPointCount() : 0,
-                    i = 0;
+            function processTelemetryObject(offset) {
+                var telemetryObject = telemetryObjects[offset],
+                    series = self.handle.getSeries(telemetryObject) || {},
+                    pointCount = series.getPointCount ? series.getPointCount() : 0;
 
-                for (; i < pointCount; i++) {
-                    rowData.push(self.table.getRowValues(telemetryObject,
-                        self.handle.makeDatum(telemetryObject, series, i)));
+                function processBatch(start, end) {
```

Comment:
The reason for using both setTimeout and $timeout is performance related. I use setTimeout to avoid triggering unnecessary digests. I had, admittedly, forgotten about the optional parameter for $timeout that skips dirty checking. I'll switch all the setTimeouts to $timeout for consistency